web-common/src/features/dashboards/dimension-table/DimensionDisplay.svelte

```diff
@@ -178,129 +113,95 @@
-      (m) => m.name === leaderboardMeasureName
-    );
-    const sortByColumnIndex = columnNames.indexOf(leaderboardMeasureName);
-    // Add comparison columns if available
-    let percentOfTotalSpliceIndex = 1;
-    if (displayComparison) {
-      percentOfTotalSpliceIndex = 2;
-      columnNames.splice(
-        sortByColumnIndex + 1,
-        0,
-        `${leaderboardMeasureName}_delta`
-      );
+  $: sortedQueryBody = prepareSortedQueryBody(
+    dimensionName,
+    $dashboardStore?.selectedMeasureNames,
+    $timeControlsStore,
+    leaderboardMeasureName,
+    $dashboardStore.dashboardSortType,
+    sortAscending,
+    filterSet
+  );
 
-      // Only push percentage delta column if selected measure is not a percentage
-      if (selectedMeasure?.format != FormatPreset.PERCENTAGE) {
-        percentOfTotalSpliceIndex = 3;
-        columnNames.splice(
-          sortByColumnIndex + 2,
-          0,
-          `${leaderboardMeasureName}_delta_perc`
-        );
-      }
-    }
-    if (validPercentOfTotal) {
-      columnNames.splice(
-        sortByColumnIndex + percentOfTotalSpliceIndex,
-        0,
-        `${leaderboardMeasureName}_percent_of_total`
-      );
+  $: sortedQuery = createQueryServiceMetricsViewComparisonToplist(
+    $runtime.instanceId,
+    metricViewName,
+    sortedQueryBody,
+    {
+      query: {
+        enabled: $timeControlsStore.ready && !!filterSet,
+      },
     }
+  );
 
-    // Make dimension the first column
-    columnNames.unshift(dimensionColumn);
-    measureNames = allMeasures.map((m) => m.name);
-
-    columns = columnNames
-      .map((columnName) => {
-        if (measureNames.includes(columnName)) {
-          // Handle all regular measures
-          const measure = allMeasures.find((m) => m.name === columnName);
-          return {
-            name: columnName,
-            type: "INT",
-            label: measure?.label || measure?.expression,
-            description: measure?.description,
-            total: referenceValues[measure.name] || 0,
-            enableResize: false,
-            format: measure?.format,
-          };
-        } else if (columnName === dimensionColumn) {
-          // Handle dimension column
-          return {
-            name: columnName,
-            type: "VARCHAR",
-            label: dimension?.label,
-            enableResize: true,
-          };
-        } else if (selectedMeasure) {
-          // Handle delta and delta_perc
-          const comparison = getComparisonProperties(
-            columnName,
-            selectedMeasure
-          );
-          return {
-            name: columnName,
-            type: comparison.type,
-            label: comparison.label,
-            description: comparison.description,
-            enableResize: false,
-            format: comparison.format,
-          };
-        }
-        return undefined;
-      })
-      .filter((column) => !!column);
-  }
+  $: tableRows = prepareDimensionTableRows(
```

Comment:
I feel like we should merge `sortedQueryBody` `sortedQuery` and `tableRows` into a single thing since they are only used at the end in `tableRows`.

Not a 100% sure how exactly, but we could perhaps do a derived store from `createQueryServiceMetricsViewComparisonToplist`. Ideally matching the new pattern using StateManagers, but that can be a further iteration.

This is not a blocker for the immediate release.

Reply:
(same as above)

Hey @AdityaHegde, regarding this comment and the other two: I'm doing more work on this stuff to enable sorting by dimension values (have a PR underway right now). Would it be ok to accept this PR for now and I can clean this up as you suggest in the follow-up PR, since I'm making changes to this code anyway?